internal/deb/extract.go

```diff
@@ -178,37 +188,26 @@ func extractData(dataReader io.Reader, options *ExtractOptions) error {
 		}
 
 		sourceIsDir := sourcePath[len(sourcePath)-1] == '/'
+		if sourceIsDir {
+			tarDirPermissions[sourcePath] = tarHeader.FileInfo().Mode()
+		}
 
 		//debugf("Extracting header: %#v", tarHeader)
 
 		var extractInfos []ExtractInfo
 		if globPath != "" {
 			extractInfos = options.Extract[globPath]
 			delete(pendingPaths, globPath)
-			if options.Globbed != nil {
-				options.Globbed[globPath] = append(options.Globbed[globPath], sourcePath)
-			}
 		} else {
 			extractInfos, ok = options.Extract[sourcePath]
 			if ok {
 				delete(pendingPaths, sourcePath)
-			} else {
-				// Base directory for extracted content. Relevant mainly to preserve
-				// the metadata, since the extracted content itself will also create
-				// any missing directories unaccounted for in the options.
-				err := options.Create(&fsutil.CreateOptions{
-					Path:        filepath.Join(options.TargetDir, sourcePath),
-					Mode:        tarHeader.FileInfo().Mode(),
-					MakeParents: true,
-				})
-				if err != nil {
-					return err
-				}
-				continue
```

Comment:
Nothing happens because `extractInfos` is empty so the for loop below, where the extraction actually happens, doesn't actually run. However, I agree that returning early makes the code more clear, I will add the `continue` back.